README.adoc

```diff
@@ -253,23 +213,94 @@ public class Album {
 }
 ----
 
-== Cloud Spanner Hibernate ORM Limitations
+=== Performance Optimizations
+
+There are some practices which can improve the execution time of Hibernate operations.
+
+==== Be Clear about Inserts or Updates
+
+Hibernate may generate additional `SELECT` statements if it is unclear whether you are attempting to insert a new record or update an existing record. The following practices can help with this:
+
+* Let Hibernate generate the ID by leaving the entity's `id` null and annotate the field with `@GeneratedValue`. Hibernate will know that the record did not exist prior if it generates a new ID. See the <<Use Generated UUIDs for ID Generation, above section>> for more details.
 
-The Cloud Spanner Hibernate Dialect supports most of the standard Hibernate and Java Persistence annotations, but there are important differences in features because of differences in https://cloud.google.com/spanner/docs/schema-and-data-model[Cloud Spanner's data model] from traditional SQL databases.
+* Or use `session.persist()` which will simply attempt the insert without generating an `ID`.
```

Comment:
I believe it will still generate and ID if needed.
```suggestion
* Or use `session.persist()` which will explicitly attempt the insert.
```

Reply:
Done.